### PR TITLE
[IMP] event, website_event: Make ics file generation available in event

### PR DESCRIPTION
**Pull request by RomainLibert**
Ics file generation should be possible even when only event is
installed as it is a basic feature of events.

Moreover we need to be able to generate such files in order to replace
the old outlook api used in the event templates, api which is not
supported anymore.

We decided to use ics files because it is supported by most calendar
schedulers hence working with outlook.

Linked to task #1853063

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/event/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class EventController(Controller):
+
+    @route(['''/event/<model("event.event", "[('state', 'in', ('confirm', 'done'))]"):event>/ics'''], type='http', auth="public")
+    def make_event_ics_file(self, event):
+        if not event or not event.registration_ids:
+            return request.not_found()
+        files = event._get_ics_file()
+        content = files[event.id]
```
Comment (jem-odoo) [inline addons/event/controllers/main.py]: As `_get_ics_file` can return an empty dict if `vobject` is not imported, i'd rather check if `files[event.id]` and return a NotFound if not.

File: addons/event/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import Controller, request, route, content_disposition
+
+
+class EventController(Controller):
+
+    @route(['''/event/<model("event.event", "[('state', 'in', ('confirm', 'done'))]"):event>/ics'''], type='http', auth="public")
+    def make_event_ics_file(self, event):
```
Comment (jem-odoo) [inline addons/event/controllers/main.py]: call it `event_ics_file`. I don't like make.

File: addons/event/models/event.py [hunk 3]
```diff
@@ -331,6 +340,31 @@ def mail_attendees(self, template_id, force_send=False, filter_func=lambda self:
...
+        """
+        result = {}
+        if not vobject:
+            return result
+
+        for event in self:
+            cal = vobject.iCalendar()
+            cal_event = cal.add('vevent')
+
+            if not event.date_begin or not event.date_end:
```
Comment (jem-odoo) [inline addons/event/models/event.py]: Those 2 fields are required I think, so we don't need to check if they are empty.

File: addons/portal/__init__.py [hunk 4]
```diff
@@ -1,6 +1,21 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+# Updating mako environement in order to be able to use slug
+try:
```
Comment (jem-odoo) [inline addons/portal/__init__.py]: just to be sure, the try/except is there in case of `mako_template_env` does not exist ?

Comment (RomainLibert) [reply addons/portal/__init__.py]: Yep exactly

**Comment by jem-odoo** [comment]
@robodoo r+ plzzz

**Pull request closed without merging**